internal/impl/postgresql/pglogicalstream/logical_stream.go

```diff
@@ -39,53 +35,72 @@ type Stream struct {
 	streamCtx    context.Context
 	streamCancel context.CancelFunc
 
-	standbyCtxCancel           context.CancelFunc
-	clientXLogPos              pglogrepl.LSN
+	standbyCtxCancel context.CancelFunc
+
+	clientXLogPos LSN
+	lsnrestart    LSN
+
 	standbyMessageTimeout      time.Duration
 	nextStandbyMessageDeadline time.Time
-	messages                   chan Wal2JsonChanges
-	snapshotMessages           chan Wal2JsonChanges
+	messages                   chan StreamMessage
 	snapshotName               string
-	changeFilter               ChangeFilter
-	lsnrestart                 pglogrepl.LSN
 	slotName                   string
 	schema                     string
 	tableNames                 []string
-	separateChanges            bool
 	snapshotBatchSize          int
+	decodingPlugin             DecodingPlugin
+	decodingPluginArguments    []string
 	snapshotMemorySafetyFactor float64
 	logger                     *service.Logger
+	monitor                    *Monitor
+	streamUncomited            bool
+	snapshotter                *Snapshotter
+	transactionAckChan         chan string
+	transactionBeginChan       chan bool
+
+	lsnAckBuffer []string
 
 	m       sync.Mutex
 	stopped bool
+
+	consumedCallback chan bool
 }
 
+// NewPgStream creates a new instance of the Stream struct
 func NewPgStream(config Config) (*Stream, error) {
 	var (
 		cfg *pgconn.Config
 		err error
 	)
 
 	sslVerifyFull := ""
-	if config.TlsVerify == TlsRequireVerify {
+	if config.TLSVerify == TLSRequireVerify {
 		sslVerifyFull = "&sslmode=verify-full"
 	}
 
-	if cfg, err = pgconn.ParseConfig(fmt.Sprintf("postgres://%s:%s@%s:%d/%s?replication=database%s",
-		config.DbUser,
-		config.DbPassword,
-		config.DbHost,
-		config.DbPort,
-		config.DbName,
+	connectionParams := ""
+	if config.PgConnRuntimeParam != "" {
+		connectionParams = "&" + config.PgConnRuntimeParam
+	}
+
+	q := fmt.Sprintf("postgres://%s:%s@%s:%d/%s?replication=database%s%s",
+		config.DBUser,
+		config.DBPassword,
```

Comment:
You mentioned something about special chars in the database password being an issue - did that get fixed?